oviewer/convert_es.go

```diff
@@ -169,193 +181,292 @@ func (es *escapeSequence) convert(st *parseState) bool {
-		case 22:
+			s.UnStrikeThrough = false
+		case 21: // Double Underline On
+			s.Underline = true // Double Underline is the same as Underline.
+			s.UnUnderline = false
+		case 22: // Bold Off
+			s.Bold = false
 			s.UnBold = true
-		case 23:
+		case 23: // Italic Off
+			s.Italic = false
 			s.UnItalic = true
-		case 24:
+		case 24: // Underline Off
+			s.Underline = false
 			s.UnUnderline = true
-		case 25:
+		case 25: // Blink Off
+			s.Blink = false
 			s.UnBlink = true
-		case 27:
+		case 27: // Reverse Off
+			s.Reverse = false
 			s.UnReverse = true
-		case 28:
-			// Invisible Off (not implemented)
-		case 29:
+		case 28: // Invisible Off
+			// (not implemented)
+		case 29: // StrikeThrough Off
+			s.StrikeThrough = false
 			s.UnStrikeThrough = true
-		case 30, 31, 32, 33, 34, 35, 36, 37:
-			s.Foreground = colorName(num - 30)
-		case 38:
-			i, color := csColor(fields[index:])
-			if i == 0 {
-				return s, nil
+		case 30, 31, 32, 33, 34, 35, 36, 37: // Foreground color
+			s.Foreground = colorName(sgr.code - 30)
+		case 38: // Foreground color extended
+			color, i, err := parseSGRColor(sgr)
+			if err != nil {
+				return OVStyle{}
 			}
 			index += i
 			s.Foreground = color
-		case 39:
+		case 39: // ForegroundColorDefault
 			s.Foreground = "default"
-		case 40, 41, 42, 43, 44, 45, 46, 47:
-			s.Background = colorName(num - 40)
-		case 48:
-			i, color := csColor(fields[index:])
-			if i == 0 {
-				return s, nil
+		case 40, 41, 42, 43, 44, 45, 46, 47: // Background color
+			s.Background = colorName(sgr.code - 40)
+		case 48: // Background color extended
+			color, i, err := parseSGRColor(sgr)
+			if err != nil {
+				return OVStyle{}
 			}
 			index += i
 			s.Background = color
-		case 49:
+		case 49: // BackgroundColorDefault
 			s.Background = "default"
-		case 53:
+		case 53: // Overline On
 			s.OverLine = true
-		case 55:
+		case 55: // Overline Off
 			s.UnOverLine = true
-		case 90, 91, 92, 93, 94, 95, 96, 97:
-			s.Foreground = colorName(num - 82)
-		case 100, 101, 102, 103, 104, 105, 106, 107:
-			s.Background = colorName(num - 92)
+		case 58: // UnderlineColor
+			// (not implemented). Increase index only.
+			_, i, err := parseSGRColor(sgr)
+			if err != nil {
+				return s
+			}
+			index += i
+		case 59: // UnderlineColorDefault
+			// (not implemented).
+		case 73, 74, 75: // VerticalAlignment
+			// (not implemented).
+		case 90, 91, 92, 93, 94, 95, 96, 97: // Bright Foreground color
+			s.Foreground = colorName(sgr.code - 82)
+		case 100, 101, 102, 103, 104, 105, 106, 107: // Bright Background color
+			s.Background = colorName(sgr.code - 92)
 		}
 	}
-	return s, nil
+	return s
 }
 
-// toESCode converts a string to an integer.
-// If the code is smaller than 0x40 for compatibility,
-// return -1 instead of an error.
-func toESCode(str string) (int, error) {
-	num, err := strconv.Atoi(str)
+// toSGRCode converts the SGR parameter to a code.
+// Supports both separators (:) and (;).
```

Comment:
You said it supports both, but I only see a split on : 🤔

Reply:
This was not a good comment.
If there was no colon(:) separated string, the paramlist (which was semicolon(;;) separated) parameter would be used.